File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = {}
        for i in range(len(nums)):
            if target - nums[i] in n:
                return [n.get(target - nums[i]), i]
            else:
                n[nums[i]] = i
        return [0,0]
                

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> used;
        for (int i = 0; i < nums.size(); i++) {
            if (used.count(target - nums[i])) {
                return {used[target - nums[i]], i};
            } else {
                used.insert({nums[i], i});
            }
        }
        return vector<int>();
    }
};